github/src/test/java/io/syndesis/github/GitHubServiceITCase.java

```diff
@@ -62,13 +68,44 @@ public void before() throws IOException {
 
         Assertions.assertThat(authToken).isNotNull().isNotBlank();
 
-        // Setup credentials for GitHub client (Token should include "repo" and "delete_repo" scopes)
-        client = new GitHubClient();
-        client.setOAuth2Token(authToken);
+        client = new KeycloakProviderTokenAwareGitHubClient();
 
         // Now that we have a client we create one of our GitHubService
         githubService = new GitHubServiceImpl(new RepositoryService(client), new UserService(client), gitWorkflow);
 
+        webserver = new DefaultMockServer();
+        webserver.start(1234);
+
+        webserver.expect().get().withPath("/auth/realms/syndesis-it/broker/github/token").andReturn(
+            200,
+            "{\"access_token\":\"" + authToken + "\",\"expires_in\":2678400,\"scope\":\"user:info user:check-access role:edit:syndesis-staging:! role:system:build-strategy-source:syndesis-staging\",\"token_type\":\"Bearer\"}"
+        )
+            .withHeader("Content-Type", "application/json")
+            .always();
+
+        KeycloakAuthenticationToken keycloakAuthenticationToken = new KeycloakAuthenticationToken(
```

Comment:
The github token is retrieved from keycloak, an authenticated principal is required to do that.